orca-core/src/main/groovy/com/netflix/spinnaker/orca/listeners/StageStatusPropagationListener.groovy

```diff
@@ -41,7 +41,9 @@ class StageStatusPropagationListener implements StageListener {
   @Override
   void afterTask(Persister persister, Stage stage, Task task, ExecutionStatus executionStatus, boolean wasSuccessful) {
     if (executionStatus) {
-      def nonBookendTasks = stage.tasks.findAll { !it.bookend }
+      def nonBookendTasks = stage.tasks.findAll {
+        !it.bookend && !it.stageStart && !it.stageEnd
```

Comment:
`stageStart` and `stageEnd` will _not_ be set in old mode. In that case it will rely on `bookend`. Ultimately that should go away.